pkg/options/patchtmpl/patch.go

```diff
@@ -41,21 +41,28 @@ type applier struct {
 	// If includeTemplates is true, applied patch templates will be included in the
 	// component objects.
 	includeTemplates bool
+
+	// Set the template missing key policy
+	missingKey string
 }
 
 // NewApplier creates a new options applier instance. The filter indicates
 // keep-only options for what subsets of patches to look for.
-func NewApplier(pt *PatcherScheme, opts *filter.Options, includeTemplates bool) options.Applier {
+func NewApplier(pt *PatcherScheme, opts *filter.Options, includeTemplates bool, missingKey string) options.Applier {
```

Comment:
rather than missingKey, how about having templateOptions string. So, the user would have to specify "missingkey=error".

- Alternatively, I'd recommend representing these as typed string consts.

type MissingKeyOpt string

const (
  Error MissingKeyOpt = "error"
  ...
) 

etc.

Such an enum would probably live one layer up.

Reply:
Ok. I just used string so I don't have to convert back to string later. But I used constants and generalized it to options.